File: app/src/main/java/org/kindone/willingtodo/pomodorocontrol/ControlRunning.java
package org.kindone.willingtodo.pomodorocontrol;

/**
 * Created by kindone on 2017. 1. 2..
 */

public class ControlRunning extends ControlState {
    public ControlRunning(ControlContext context) {
        super(context);
    }

    @Override
    public void start() {
        // DO NOTHING
    }

    @Override
    public void pause() {
        changeState(new ControlPaused(mContext));
    }

    @Override
    public void resume() {
        // DO NOTHING
    }

    @Override
    public void stop() {
        changeState(new ControlStopped(mContext));
    }

    @Override
    public int getMode() {
        return MODE_PAUSED;
    }
}

File: app/src/main/java/org/kindone/willingtodo/pomodorocontrol/ControlRunning.kt
package org.kindone.willingtodo.pomodorocontrol

/**
 * Created by kindone on 2017. 1. 2..
 */

class ControlRunning(context: ControlContext) : ControlState(context) {

    override fun start() {
        // DO NOTHING
    }

    override fun pause() {
        changeState(ControlPaused(mContext))
    }

    override fun resume() {
        // DO NOTHING
    }

    override fun stop() {
        changeState(ControlStopped(mContext))
    }

    override fun getMode(): Int {
        return ControlState.MODE_PAUSED
    }
}
